runtime/src/test/java/com/dylibso/chicory/runtime/MStackTest.java

```diff
@@ -0,0 +1,91 @@
+package com.dylibso.chicory.runtime;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.dylibso.chicory.wasm.types.Value;
+import org.junit.jupiter.api.Test;
+
+public class MStackTest {
```

Comment:
@bhelx you mentioned to regret calling this `MStack` do you have an alternative name proposal?

Reply:
`OperandStack` is the proper name IMHO. At least this will clearly distinguish it from CallStack.

Reply:
`MStackTest` is kinda of testing some minor logic inside the MStack, like throwing appropriate exception etc. It would be much easier to replace in future if needed, b/c each time running the whole integration tests suite is rather time-consuming.

Reply:
Not specifically. Something like `ValueStack` or `OperandStack` both make sense to me because that's kind of how we use it at the moment. In the spec, the stack is used for more than operands though. But I'm not sure if we want to put labels and activations on the stack so maybe it will always only be values for us. https://www.w3.org/TR/wasm-core-1/#stack

What do you think @andreaTP ? Naming is obviously one of my weak spots :)

Reply:
> like throwing appropriate exception etc.

This is covered by the testsuite.

> It would be much easier to replace in future if needed

Happy to re-evaluate when necessary, in the future, we don't want to introduce complexity and code to be maintained if it's not useful with the current implementation.

> Naming is obviously one of my weak spots

Same here :sweat_smile: what about `WasmStack`?
No super-strong opinions, though!

Reply:
IDK WasmStack feels a bit too generic for me. It does feel like right now it is a stack for holding operands. thus `OperandStack` makes sense. I recommended `ValueStack` because it holds `Value`s (both the chicory type and the wasm type). So it feels closer to the code. Although we instructions do have operands then name of the property is `operands()`, they manifest in the code as `Value`s.

I don't like MStack because it doesn't mean anything. I literally just didn't want to conflict with `Stack` which i was using at the time 😆 . I think in my head it was `M(achine)Stack`.

Reply:
Okay since we're kind of not coming to a conclusion and we don't have strong opinions, i'm just gonna say `OperandStack` is good for us

Reply:
Not fussy, but `Operands` conflicts with the notion of "immediate values" that we call `Operands`.
If we call the Stack `OperandStack` I suggest we rename `operands` used in the machine to something different.

To be honest, `M(achine)Stack` looks like the best option to me.